app/core/memberships/endpoints_memberships.py

```diff
@@ -0,0 +1,499 @@
+    if db_user is None:
+        raise HTTPException(status_code=404, detail="User not found")
+
+    db_user_membership = schemas_memberships.UserMembershipSimple(
+        id=uuid.uuid4(),
+        user_id=user_id,
+        association_membership_id=user_membership.association_membership_id,
+        start_date=user_membership.start_date,
+        end_date=user_membership.end_date,
+    )
+    await validate_user_membership(db_user_membership, db)
+
+    cruds_memberships.create_user_membership(db=db, user_membership=db_user_membership)
+    try:
+        await db.commit()
+    except Exception:
+        raise HTTPException(
+            status_code=500,
+            detail="Failed to create user membership",
+        )
+    return schemas_memberships.UserMembershipComplete(
+        **db_user_membership.__dict__,
+        user=schemas_users.CoreUserSimple(
+            name=db_user.name,
+            id=db_user.id,
+            firstname=db_user.firstname,
+            nickname=db_user.nickname,
+            account_type=db_user.account_type,
+            school_id=db_user.school_id,
+        ),
+    )
+
+
+@router.post(
+    "/memberships/{association_membership_id}/add-batch/",
+    status_code=201,
+    response_model=list[schemas_memberships.MembershipUserMappingEmail],
+)
+async def add_batch_membership(
+    association_membership_id: uuid.UUID,
+    memberships_details: list[schemas_memberships.MembershipUserMappingEmail],
+    db: AsyncSession = Depends(get_db),
+    user: models_users.CoreUser = Depends(is_user_in(GroupType.admin)),
+):
+    """
+    Add a batch of user to a membership.
+
+    Return the list of unknown users whose email is not in the database.
+
+    **User must be an administrator to use this endpoint.**
+    """
+    db_association_membership = (
+        await cruds_memberships.get_association_membership_by_id(
+            db=db,
+            membership_id=association_membership_id,
+        )
+    )
+    if db_association_membership is None:
+        raise HTTPException(
+            status_code=400,
+            detail="Association membership not found",
+        )
+
+    unknown_users: list[schemas_memberships.MembershipUserMappingEmail] = []
+    for detail in memberships_details:
+        detail_user = await cruds_users.get_user_by_email(
+            db=db,
+            email=detail.user_email,
+        )
+        if not detail_user:
+            unknown_users.append(detail)
+            continue
+        stored_memberships = await cruds_memberships.get_user_memberships_by_user_id_and_association_membership_id(
+            db=db,
+            user_id=detail_user.id,
+            association_membership_id=association_membership_id,
+        )
+        if not any(
+            stored_membership.start_date == detail.start_date
+            and stored_membership.end_date == detail.end_date
+            for stored_membership in stored_memberships
+        ):
```

Comment:
Why don't you just use a crud `get_user_memberships_by_user_id_start_end_and_association_membership_id`?